pkg/util/retry/retry.go

```diff
@@ -69,6 +73,18 @@ func Do(ctx context.Context, fn func() error, opts ...Option) error {
 
 // errUnrecoverable is error instance for unrecoverable.
 var errUnrecoverable = errors.New("unrecoverable error")
+var errContextDone = errors.New("context done error")
+
+// ContextDone method wrap an error to context done error. This will make retry
+// quick return.
+func ContextDone(err error) error {
+	return merr.Combine(err, errContextDone)
+}
+
+// IsContextDone is used to judge whether the error is wrapped by errContextDone.
+func IsContextDone(err error) bool {
```

Comment:
`merr.IsCanceledOrTimeout` may be what you are finding, simply combining the errors with `ctx.Err()` should work well